### PR TITLE
No face detected

### PR DESCRIPTION
**Pull request by AndreaMacarulla**

### CODE REVIEW

File: lr_face/models.py [hunk 1]
```diff
@@ -90,7 +90,14 @@ def predict_proba(self, X: List[FacePair]) -> np.ndarray:
...
             embedding2 = self.embedding_model.embed(pair.second, cache_dir)
-            score = np.linalg.norm(embedding1 - embedding2)
+            if embedding1 is None or embedding2 is None:
+                # If face was not detected, score = -1
+                score = -1
+                scores.append([score, 1 - score])
+            else:
+                score = spatial.distance.cosine(embedding1, embedding2)
+                scores.append([score, 1 - score])
+            # score = np.linalg.norm(embedding1 - embedding2)
```
Comment (judithvdw) [inline lr_face/models.py:100]: is there a reason you are now using spatial.distance.cosine in stead off np.linalg.norm?

File: lr_face/models.py [hunk 2]
```diff
@@ -79,11 +85,19 @@ def predict_proba(self, X: List[FacePair]) -> np.ndarray:
...
+        rm_pair = []
         cache_dir = EMBEDDINGS_DIR
         for pair in X:
             embedding1 = self.embedding_model.embed(pair.first, cache_dir)
             embedding2 = self.embedding_model.embed(pair.second, cache_dir)
-            score = spatial.distance.cosine(embedding1, embedding2)
+            if embedding1 is None or embedding2 is None:
+                # If face was not detected, score = -1
+                score = -1
+                scores.append([score, 1 - score])
```
Comment (rolfypma) [inline lr_face/models.py:96]: this line is executed in both the if and the else, and can thus be moved to below

File: preprocessing.py [hunk 3]
```diff
@@ -50,27 +52,40 @@ def run(input_folder, output_folder, recursive):
     paths = filter(is_image, paths)
     meta = defaultdict(list)
     for path in paths:
+        print(path)
```
Comment (rolfypma) [inline preprocessing.py:55]: debug code?

Comment (AndreaMacarulla) [reply preprocessing.py:55]: yes XD

**Pull request merged**